pkg/apis/mysql/v1alpha1/types.go

```diff
@@ -0,0 +1,419 @@
+	Executor *BackupExecutor `json:"executor"`
+
+	// StorageProvider is the configuration of where and how backups should be stored.
+	StorageProvider *BackupStorageProvider `json:"storageProvider"`
+
+	// Cluster is a reference to the Cluster to which the Backup belongs.
+	Cluster *corev1.LocalObjectReference `json:"cluster"`
+
+	// AgentScheduled is the agent hostname to run the backup on.
+	// TODO(apryde): ScheduledAgent (*corev1.LocalObjectReference)?
+	AgentScheduled string `json:"agentscheduled"`
+}
+
+// BackupExecutor represents the configuration of the tool performing the backup. This includes the tool
+// to use, and, what database and tables should be backed up.
+// The storage of the backup is configured in the relevant Storage configuration.
+type BackupExecutor struct {
+	// Name of the tool performing the backup, e.g. mysqldump.
+	Name string `json:"name"`
+	// Databases are the databases to backup.
+	Databases []string `json:"databases"`
+}
+
+// BackupStorageProvider defines the configuration for storing a MySQL backup to a storage service.
+// The generation of the backup is configured in the Executor configuration.
+type BackupStorageProvider struct {
+	// Name denotes the type of storage provider that will store and retrieve the backups,
+	// e.g. s3, oci-s3-compat, aws-s3, gce-s3, etc.
+	Name string `json:"name"`
+	// SecretRef is a reference to the Kubernetes secret containing the configuration for uploading
+	// the backup to authenticated storage.
+	SecretRef *corev1.LocalObjectReference `json:"secretRef,omitempty"`
+	// Config is generic string based key-value map that defines non-secret configuration values for
+	// uploading the backup to storage w.r.t the configured storage provider.
+	Config map[string]string `json:"config,omitempty"`
+}
+
+// BackupPhase represents the current life-cycle phase of a Backup.
+type BackupPhase string
+
+const (
+	// BackupPhaseUnknown means that the backup hasn't yet been processed.
+	BackupPhaseUnknown BackupPhase = ""
+
+	// BackupPhaseNew means that the Backup hasn't yet been processed.
+	BackupPhaseNew BackupPhase = "New"
+
+	// BackupPhaseScheduled means that the Backup has been scheduled on an
+	// appropriate replica.
+	BackupPhaseScheduled BackupPhase = "Scheduled"
+
+	// BackupPhaseStarted means the backup is in progress.
+	BackupPhaseStarted BackupPhase = "Started"
+
+	// BackupPhaseComplete means the backup has terminated successfully.
+	BackupPhaseComplete BackupPhase = "Complete"
+
+	// BackupPhaseFailed means the backup has terminated with an error.
+	BackupPhaseFailed BackupPhase = "Failed"
+)
+
+// BackupOutcome describes the location of a MySQL Backup
+type BackupOutcome struct {
+	// Location is the Object Storage network location of the Backup.
+	Location string `json:"location"`
+}
+
+// BackupStatus captures the current status of a MySQL backup.
+type BackupStatus struct {
+	// Phase is the current life-cycle phase of the Backup.
+	Phase BackupPhase `json:"phase"`
```

Comment:
Replace with conditions?

Reply:
+1

https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#typical-status-properties

> Newer API types should use conditions instead. 

You should avoid using phase as it's a deprecated pattern in core.